daphne_server/src/router/mod.rs

```diff
@@ -0,0 +1,455 @@
+    ) -> Result<Self, Self::Rejection> {
+        #[derive(Debug, Deserialize)]
+        #[serde(deny_unknown_fields)]
+        struct PathParams {
+            version: DapVersion,
+            #[serde(
+                default,
+                deserialize_with = "daphne::messages::base64url::deserialize_opt"
+            )]
+            task_id: Option<TaskId>,
+            #[serde(
+                default,
+                deserialize_with = "daphne::messages::base64url::deserialize_opt"
+            )]
+            agg_job_id: Option<AggregationJobId>,
+            #[serde(
+                default,
+                deserialize_with = "daphne::messages::base64url::deserialize_opt"
+            )]
+            collect_job_id: Option<CollectionJobId>,
+        }
+
+        let (mut parts, body) = req.into_parts();
+        let Path(PathParams {
+            version,
+            task_id,
+            agg_job_id,
+            collect_job_id,
+        }) = Path::from_request_parts(&mut parts, state)
+            .await
+            .map_err(|e| (StatusCode::BAD_REQUEST, e.to_string()))?;
+
+        let sender_auth = DaphneAuth {
+            bearer_token: parts
+                .headers
+                .get("DAP-Auth-Token")
+                .and_then(|v| v.to_str().ok())
+                .map(BearerToken::from),
+            // TODO(mendess) figure out tls
+            cf_tls_client_auth: None,
+        };
+
+        let media_type = DapMediaType::from_str_for_version(
+            version,
+            parts
+                .headers
+                .get(CONTENT_TYPE)
+                .and_then(|v| v.to_str().ok()),
+        );
+
+        let taskprov = parts
+            .headers
+            .get("dap-taskprov")
+            .and_then(|v| v.to_str().ok())
+            .map(String::from);
+
+        // TODO(mendess): this is very eager, we could redesign DapResponse later to allow for
+        // streaming of data.
+        let payload = body
+            .into_data_stream()
+            .try_fold(Vec::new(), |mut buf, bytes| async move {
+                buf.extend_from_slice(&bytes);
+                Ok(buf)
+            })
+            .await;
+        let payload = match payload {
+            Ok(payload) => payload,
+            Err(e) => return Err((StatusCode::INTERNAL_SERVER_ERROR, e.to_string())),
+        };
+
+        let (task_id, resource) = match version {
+            DapVersion::Draft02 => {
+                // Parse the task ID from the front of the request payload and use it to look up the
+                // expected bearer token.
+                //
+                // TODO(cjpatton) Add regression tests that ensure each protocol message is prefixed by the
+                // task ID.
+                //
+                // TODO spec: Consider moving the task ID out of the payload. Right now we're parsing it
+                // twice so that we have a reference to the task ID before parsing the entire message.
```

Comment:
No longer relevant
```suggestion
```

Reply:
I had a feeling it wasn't relevant, but wasn't 100% sure.